dask/dataframe/io/tests/test_parquet.py

```diff
@@ -2216,15 +2216,15 @@ def test_pathlib_path(tmpdir, engine):
     assert_eq(ddf, ddf2)
 
 
-@PYARROW_MARK
+@PYARROW_LE_MARK
```

Comment:
There are several changes to seemingly unrelated tests. I'm curious why these were needed

Reply:
Yes, good catch.  These test changes are mostly unrelated to logic added to this PR.  However, this PR does add a `dataset_kwargs` check to the pyarrow engine to avoid cases where we are silently ignoring user-privided options.  This small change triggered some CI failures that were mostly related to legacy-vs-dataset behavior.

Reply:
Hmm does that mean the `dataset=` is supported for the `pyarrow-legacy` engine? Naively I thought it was only supported when using the `pyarrow-dataset` engine

Reply:
That's right. The `dataset=` option was introduced before `pyarrow.dataset` existed. It was meant to mean "pass these kwargs to `ParquetDataset`".